### PR TITLE
ARROW-7567: [Java] Bump Checkstyle from 6.19 to 8.19

### PR DESCRIPTION
**Pull request by Fokko**
Bump checkstyle to the latest version, and fix the new violations based on new rules that have been introduced.

### CODE REVIEW

**Comment by bkietz** [comment]
@Fokko could you assign yourself on JIRA?

**Comment by Fokko** [comment]
I'd love to, but I don't have sufficient rights.

**Comment by wesm** [comment]
I added Fokko as a contributor and assigned several issues

**Comment by Fokko** [comment]
Thanks @wesm

File: java/dev/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -94,9 +81,6 @@
             <property name="tokens" value="LITERAL_TRY, LITERAL_FINALLY, LITERAL_IF, LITERAL_ELSE, LITERAL_SWITCH"/>
         </module>
         <module name="NeedBraces"/>
-        <module name="LeftCurly">
-            <property name="maxLineLength" value="100"/>
```
Comment (emkornfield) [inline java/dev/checkstyle/checkstyle.xml]: why this removal?

File: java/vector/src/test/java/org/apache/arrow/vector/ipc/TestJSONFile.java [hunk 2]
```diff
@@ -300,30 +296,25 @@ public void testWriteReadDecimalJSON() throws IOException {
...
     ) {
-      JsonFileReader reader = new JsonFileReader(file, readerAllocator);
       Schema schema = reader.start();
       LOGGER.debug("reading schema: " + schema);
 
       // initialize vectors
       try (VectorSchemaRoot root = reader.read();) {
         validateDecimalData(root);
       }
-      reader.close();
```
Comment (emkornfield) [inline java/vector/src/test/java/org/apache/arrow/vector/ipc/TestJSONFile.java]: does this really fail on style or is this just separate cleanup?

**Comment by emkornfield** [comment]
@Fokko thanks a couple of minor comments otherwise LGTM

File: java/dev/checkstyle/checkstyle.xml (hunk 1 above)
Comment (Fokko) [reply java/dev/checkstyle/checkstyle.xml]: This has been removed in 8.2: https://checkstyle.sourceforge.io/releasenotes.html#Release_8.2 Third point in the list.

File: java/vector/src/test/java/org/apache/arrow/vector/ipc/TestJSONFile.java (hunk 2 above)
Comment (Fokko) [reply java/vector/src/test/java/org/apache/arrow/vector/ipc/TestJSONFile.java]: The reader is now closed by the try-with-resources pattern: https://github.com/apache/arrow/pull/6186/files#diff-1ed18d6881a6b20dd9f67c9878605613R309

A lot of the readers in this file weren't being closed, so I've moved them to the try-with-resources block.

**Comment by emkornfield** [comment]
+1 thanks.

**Pull request closed without merging**